Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1566,6 +1639,84 @@ void UpdateReference(EntityRecord dst, EntityRecord src,MeshSyncPlayerConfig con
         }
     }
 
+    EntityRecord UpdateInstancedEntity(TransformData data)
+    {
+        var config = GetConfigV();
+        
+        if (!m_clientInstancedEntities.TryGetValue(data.path, out EntityRecord rec)) {
+            var trans = 
+                FilmInternalUtilities.GameObjectUtility.FindOrCreateByPath(m_rootObject, data.path, false);
+            
+            rec = new EntityRecord {
+                go = trans.gameObject,
+                trans = trans,
+                recved = true,
+            };
+            
+            m_clientInstancedEntities.Add(data.path, rec);
+        }
+        
+        UpdateTransformEntity(data, config, rec);
+        UpdateMeshEntity((MeshData)data, config, rec);
+
+        m_clientInstancedEntities[data.path] = rec;
+        
+        // If there is a instance info record that references the new object
+        if (m_clientInstances.TryGetValue(data.path, out InstanceInfoRecord infoRecord)) {
+            // Update the renderer reference
+            infoRecord.go = rec.go;
+            var renderer = infoRecord.renderer;
+            if (renderer != null) {
+                renderer.UpdateReference(rec.go);
+            }
+        }
+
+        return rec;
+    }
+    
+    InstanceInfoRecord UpdateInstanceInfo(InstanceInfoData data)
+    {
+        InstanceInfoRecord infoRecord = null;
+        
+        var path = data.path;
+        
+        if (!m_clientInstances.TryGetValue(path, out infoRecord)) {
+            infoRecord = new InstanceInfoRecord();
+            m_clientInstances.Add(path, infoRecord);
+        }
+        
+        // Look for reference in client instanced objects
+        if (m_clientInstancedEntities.TryGetValue(path, out EntityRecord instancedEntityRecord)) { 
+            infoRecord.go = instancedEntityRecord.go;
+        }
+        else if (m_clientObjects.TryGetValue(path, out EntityRecord entityRecord)) {
+            infoRecord.go = entityRecord.go;
+        }
+        else {
+            throw new Exception($"[MeshSync] No instanced entity found for path {data.path}");
+        } 
+        
+        // Look for parent in client objects
+        if (m_clientObjects.TryGetValue(data.parentPath, out EntityRecord parentRecord)) {
+            if (infoRecord.renderer == null) {
+                infoRecord.renderer = parentRecord.go.AddComponent<MeshSyncInstanceRenderer>();
```

Comment:
I am wondering if we should use `GetOrAddComponent()` here

Reply:
A parent might have multiple rendering components for different instantiated objects, so we must use AddComponent always